packages/backend-common/src/config.ts

```diff
@@ -51,53 +66,69 @@ class ObservableConfigProxy implements Config {
     };
   }
 
+  private select(required: true): Config;
+  private select(required: false): Config | undefined;
+  private select(required: boolean): Config | undefined {
+    if (this.parent && this.parentKey) {
+      if (required) {
+        return this.parent.select(true).getConfig(this.parentKey);
+      }
+      return this.parent.select(false)?.getOptionalConfig(this.parentKey);
+    }
+
+    return this.config;
+  }
+
   has(key: string): boolean {
-    return this.config.has(key);
+    return this.select(false)?.has(key) ?? false;
   }
   keys(): string[] {
-    return this.config.keys();
+    return this.select(false)?.keys() ?? [];
   }
```

Comment:
Hm not sure I understand why this one is false.

Reply:
Same logic as in reader, if there's no backing data it returns `[]` https://github.com/backstage/backstage/blob/2325563877af4b21b1e4f9cc1e1fc8628072728d/packages/config/src/reader.ts#L107-L109

Reply:
Ahh yeah there's no optionalKeys